### PR TITLE
Remove play-enhancer from docs, project is dead

### PR DESCRIPTION
**Pull request by mkurz**
Project was archived: https://github.com/playframework/play-enhancer
I think it does not work correctly anymore with newer JDKs.
play-ebean also removed it already: https://github.com/playframework/play-ebean/issues/201 (PR https://github.com/playframework/play-ebean/pull/203)

### CODE REVIEW

File: documentation/manual/working/javaGuide/main/forms/JavaForms.md [hunk 1]
```diff
@@ -1,8 +1,6 @@
 <!--- Copyright (C) Lightbend Inc. <https://www.lightbend.com> -->
 # Handling form submission
 
-Before you start with Play forms, read the documentation on the [[Play enhancer|PlayEnhancer]]. The Play enhancer generates accessors for fields in Java classes for you, so that you don't have to generate them yourself. You may decide to use this as a convenience. All the examples below show manually writing accessors for your classes.
```
Comment (mkurz) [inline documentation/manual/working/javaGuide/main/forms/JavaForms.md:4]: > All the examples below show manually writing accessors for your classes.

That statement is not true  anymore, all examples below have explicit getters and setter.

File: documentation/build.sbt [hunk 2]
```diff
@@ -16,7 +15,6 @@ val DocsApplication = config("docs").hide
 
 lazy val main = Project("Play-Documentation", file("."))
   .enablePlugins(PlayDocsPlugin, SbtTwirl)
-  .disablePlugins(PlayEnhancer)
```
Comment (mkurz) [inline documentation/build.sbt:19]: We add the plugin, but we disabled it. Funny.

**Review by gmethvin — APPROVED** [review]

**Comment by PromanSEW** [comment]
This must be backported to `2.8.x`

**Comment by mkurz** [comment]
@Mergifyio backport 2.8.x

**Comment by mkurz** [comment]
@PromanSEW Can you create that backport pull request? Thanks!

**Pull request merged**